src/app/components/chefs-table/chefs-table-datasource.ts

```diff
@@ -0,0 +1,65 @@
+import { DataSource } from '@angular/cdk/collections';
+import { MatPaginator } from '@angular/material/paginator';
+import { MatSort } from '@angular/material/sort';
+import { map } from 'rxjs/operators';
+import { Observable, of as observableOf, merge } from 'rxjs';
+import { IChef } from '../../interfaces/data.interface';
+import { ChefService } from '../../services/chef.service';
+
+export class ChefsTableDataSource extends DataSource<IChef> {
+  data: IChef[] = [];
+  paginator: MatPaginator | undefined;
+  sort: MatSort | undefined;
+
+  constructor(private chefService: ChefService) {
+    super();
+  }
+
+  fetchData() {
+    this.chefService.getChefs().subscribe((data) => {
+      this.data = data;
+    })
+  }
+
+  connect(): Observable<IChef[]> {
+    this.fetchData();
+    if (this.paginator && this.sort) {
+      return merge(observableOf(this.data), this.paginator.page, this.sort.sortChange)
+        .pipe(map(() => {
+          return this.getPagedData(this.getSortedData([...this.data]));
+        }));
+    } else {
+      throw Error('Please set the paginator and sort on the data source before connecting.');
+    }
+  }
+
+  disconnect(): void { }
+
+  private getPagedData(data: IChef[]): IChef[] {
+    if (this.paginator) {
+      const startIndex = this.paginator.pageIndex * this.paginator.pageSize;
+      return data.splice(startIndex, this.paginator.pageSize);
+    } else {
+      return data;
+    }
+  }
+
+  private getSortedData(data: IChef[]): IChef[] {
+    if (!this.sort || !this.sort.active || this.sort.direction === '') {
```

Comment:
Try this instead:
if (!this.sort?.active || this.sort?.direction === '') 

Read into "optional chaining" on google